test/test_processing_utils.py

```diff
@@ -434,5 +445,23 @@ def test_local_urls_fail(url):
         "https://huggingface.co/datasets/dylanebert/3dgs/resolve/main/luigi/luigi.ply",
     ],
 )
-def test_public_urls_pass(url):
-    assert processing_utils.get_public_url(url)
+async def test_public_urls_pass(url):
+    await processing_utils.validate_url(url)
+
+
+@pytest.mark.asyncio
+async def test_public_request_pass():
+    tempdir = tempfile.TemporaryDirectory()
+    assert await processing_utils.ssrf_protected_httpx_download(
+        "https://en.wikipedia.org/static/images/icons/wikipedia.png", tempdir.name
+    )
+    assert os.path.exists(os.path.join(tempdir.name, "wikipedia.png"))
+
+
+@pytest.mark.asyncio
+async def test_private_request_fail():
+    with pytest.raises(ValueError, match="Unable to resolve"):
+        tempdir = tempfile.TemporaryDirectory()
+        await processing_utils.ssrf_protected_httpx_download(
+            "http://192.168.1.250.nip.io/image.png", tempdir.name
```

Comment:
Just curious, did you set up this url yourself?

Reply:
nip.io is free service that is a "DNS wildcard" that will redirect to the ip address in the subdomain. The subdomain ip address is just the local network.

Reply:
Neat